geopandas/io/file.py

```diff
@@ -103,6 +104,24 @@ def _check_pyogrio(func):
         )
 
 
+def _check_metadata_supported(metadata: str | None, engine: str, driver: str) -> None:
```

Comment:
The original PR had these checks in the fiona and pyogrio branches separately, I think this minimises added noise there, and makes things clearer to update if we ever have more drivers which support metadata.